core/src/test/java/org/infinispan/statetransfer/StateTransferRestart2Test.java

```diff
@@ -0,0 +1,154 @@
+package org.infinispan.statetransfer;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+
+import org.infinispan.Cache;
+import org.infinispan.commands.ReplicableCommand;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.remoting.inboundhandler.DeliverOrder;
+import org.infinispan.remoting.responses.Response;
+import org.infinispan.remoting.rpc.ResponseFilter;
+import org.infinispan.remoting.rpc.ResponseMode;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.remoting.transport.jgroups.JGroupsTransport;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.fwk.CleanupAfterMethod;
+import org.infinispan.test.fwk.TransportFlags;
+import org.infinispan.transaction.lookup.DummyTransactionManagerLookup;
+import org.jgroups.protocols.DISCARD;
+import org.testng.annotations.Test;
+
+/**
+ * Tests scenario for ISPN-7127
+ *
+ * - create nodes A, B - start node C - starts state transfer from B to C
+ * - abruptly kill B before it is able to reply to the StateRequestCommand from C
+ * - C resends the request to A
+ * - finally cluster A, C is formed where all entries are properly backed up on both nodes
+ *
+ * @author Michal Linhard
+ * @author Dan Berindei
+ * @since 5.2
+ */
+@Test(groups = "functional", testName = "statetransfer.StateTransferRestartTest")
+@CleanupAfterMethod
+public class StateTransferRestart2Test extends MultipleCacheManagersTest {
+
+   private ConfigurationBuilder cfgBuilder;
+
+   private void waitForStateTransfer(Cache... caches) throws InterruptedException {
+      StateTransferManager[] stm = new StateTransferManager[caches.length];
+      for (int i = 0; i < stm.length; i++) {
+         stm[i] = TestingUtil.extractComponent(caches[i], StateTransferManager.class);
+      }
+      while (true) {
+         boolean inProgress = false;
+         for (StateTransferManager aStm : stm) {
+            if (aStm.isStateTransferInProgress()) {
+               inProgress = true;
+               break;
+            }
+         }
+         if (!inProgress) {
+            break;
+         }
+         wait(100);
```

Comment:
I guess it never actually got to this point :) I'm removing this method and using `TestingUtil.waitForRehashToComplete()` instead.